pkg/controller/utils/utils.go

```diff
@@ -299,3 +300,23 @@ func GetExpectedTyphaScale(nodes int) int {
 	}
 	return typhas
 }
+
+// GetElasticLicenseType returns the license type from elastic-licensing ConfigMap that ECK operator keeps updated.
+func GetElasticLicenseType(ctx context.Context, cli client.Client, logger logr.Logger) (render.ElasticLicenseType, error) {
+	cm := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{Name: render.ECKLicenseConfigMapName, Namespace: render.ECKOperatorNamespace}, cm)
+	if err != nil {
+		// If ConfigMap is not available, it means ECK operator is not running yet, so log the information and proceed
+		if errors.IsNotFound(err) {
```

Comment:
You don't need / shouldn't have a check here for is not found, you need to return an error so that we don't progress, so just return the error as is

Reply:
This function is called in two controllers and we need this check to differentiate 2 scenario.
- in intrusion-detection, if elastic-licensing is not found or if there is an from this function, we don't install ml jobs
- in logstorage, if elastic-licensing is not found we need to proceed with rendering ECK operator related resource, if there is any other error in `GetElasticLicenseType` we don't progress

Reply:
I think you should just check if the error is a not found one in the operator and continue, and just leave this simple. As far as this function is concerned we should consider not found an error, and if the caller would like to ignore that error then it can do so but checking if it's a not found error.

The code you have of `err != nil || elasticLicenseType == render.ElasticLicenseTypeUnknown` should really just be `err != nil && errors.IsNotFound(err)` (always return Unkown if there's an error).